vspace/src/main/java/edu/asu/diging/vspace/web/staff/ExhibitionConfigurationController.java

```diff
@@ -60,26 +61,33 @@ public String showExhibitions(Model model) {
      */
     @RequestMapping(value = "/staff/exhibit/config", method = RequestMethod.POST)
     public RedirectView createOrUpdateExhibition(HttpServletRequest request,
-            @RequestParam(required = false, name = "exhibitionParam") String exhibitID,
-            @RequestParam("spaceParam") String spaceID, @RequestParam("title") String title, RedirectAttributes attributes)
-            throws IOException {
+        @RequestParam(required = false, name = "exhibitionParam") String exhibitID,
+        @RequestParam("spaceParam") String spaceID, @RequestParam("title") String title,
+        @RequestParam("exhibitMode") ExhibitionModes exhibitMode,
+        @RequestParam(value = "customMessage", required = false, defaultValue = "") String customMessage,
+        RedirectAttributes attributes) throws IOException {
 
         ISpace startSpace = spaceManager.getSpace(spaceID);
 
         Exhibition exhibition;
-        if (exhibitID == null || exhibitID.isEmpty()) {
+        if(exhibitID==null || exhibitID.isEmpty()) {
             exhibition = (Exhibition) exhibitFactory.createExhibition();
         } else {
             exhibition = (Exhibition) exhibitManager.getExhibitionById(exhibitID);
         }
-
         exhibition.setStartSpace(startSpace);
         exhibition.setTitle(title);
+        exhibition.setMode(exhibitMode);
+        if(exhibitMode.equals(ExhibitionModes.OFFLINE) && !(customMessage.equals("")
+            || customMessage.equals(ExhibitionModes.OFFLINE.getValue()))) {
+            exhibition.setCustomMessage(customMessage);
+        } else {
+            exhibition.setCustomMessage("");
```

Comment:
why don't you leave that with the default (`null`) in this case?

Reply:
I wanted to reset the message in case the exhibition mode is changed to Active / maintenance from Offline. Should I store the old message even after the offline mode is removed ? In that case, it will show up in the config page while setting the exhibition to Offline at a later stage.

Reply:
I would keep it. Chances are that the maintenance message won't change much (once the user settled on one they would probably reuse it for the next time).